docs/DeveloperGuide.md

```diff
@@ -2,28 +2,454 @@
+      [description] [$amount] `. This will match when the optional flags are not included. An empty string will then be
+      returned.
+    - `-(in|out)\s+(.+)\$([\d.]+)\s+(.*)` checks for the corresponding structure `[-in/out] [whitepsace]
+      [description] [$amount] [remaining input]`. This will match when at least one of the optional flags are included,
+      and a string corresponding to `[remaining input]` will be returned.
+
+
+2. If an empty string is returned, an addCommand object will be returned from function addStatement. Otherwise,
+   a variable `String remainingInformation` will correspond to `[remaining input]` above. `remainingInformation` will
+   then
+   be checked for if it contains `-c` or `-date` flags.
+
+
+3. if `remainingInformation` contains a `-c` flag, it means that the user has provided a category for the item.
+   Thus, it will be passed into the function setCategory. setCategory will then use the following regex to match
+   `remainingInformation`:
+    - `-c\\s+(\\S+)` checks for the corresponding structure `[-c] [whitespace] [1-word category]`.
+    - `-c\\s+(\\S+)\\s+(.*)` checks for the same thing except category can be multiple words
+
+4. The last field to check is the `-date` field. If present, the setDate function will be called
+   on `remainingInformation`.
+   The setDate function will then use the following regex to match.
+    - `-date\\s+(\\d{2}/\\d{2}/\\d{4})` matches the corresponding
+      structure`[-date] [whitespace] [date in DD/MM/YYYY format]`
+
+These steps will ultimately parse the user's input and extract the necessary information. If any pattern does not match,
+our parser will throw a `RainDayException` indicating the wrong input format.
+
+#### Alternatives considered
+
+We have tried using the .split() function of strings to extract information. However, as we introduced more features and
+the commands grew more complex, the .split() function became very messy and inconvenient. Using this will also result in
+us needing to do more exceptions and error handling, which just made the entire process very complicated. Thus, we opted
+to use regular expressions, which is a more tidy and logical way to parse the inputs.
+
+### Viewing your data `view`
+
+- The command `view` is used to view all statements, and a ViewCommand object will be created.
+    - The format for the command is `view TIMESPAN -sort`
+    - Any other characters after any valid view command are automatically ignored
+- Information is presented in a table format to help improve clarity for users.
+    - The table includes information in the summary such as whether it is sorted and the amount of history shown. 
+    - This was deliberate as presenting this information at the bottom makes it easier for users to spot, 
+      as placing them at the top may cause users to miss it if they have a large table
+
+![ViewCommandSequenceDiagram.png](images\DeveloperGuide\ViewCommandSequenceDiagram.png)
```

Comment:
`sortByValue` and `Convert to Valid String` should have another activation bar to represent them.

![image](https://user-images.githubusercontent.com/85627680/228767959-81404c24-4b93-463f-b7c7-99f522ca5a24.png)